tensorrt_llm/_torch/auto_deploy/utils/node_utils.py

```
@@ -483,12 +545,155 @@ def successors(
+        queue = list(sources)
+        start_nodes = set(sources)
+        attr_next = "users"
+    elif sinks is not None:
+        # case 3
+        assert boundary_condition is not None, "boundary_condition must be provided for case 3"
+        # Initialize queue with sinks and mark them as seen
+        queue = list(sinks)
+        start_nodes = set(sinks)
+        attr_next = "args"
+    else:
+        raise ValueError("Either sinks or sources must be provided")
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)